src/OpenTelemetry/Trace/BatchingActivityProcessor.cs

```diff
@@ -148,31 +148,34 @@ public override void OnEnd(Activity activity)
                 return;
             }
 
-            var size = Interlocked.Increment(ref this.currentQueueSize);
+            if (activity.Recorded)
```

Comment:
i think this check should be the very 1st thing in this method. We dont want to log about queue full for an activity which isn't going to affect or be affected by the queue size.